### PR TITLE
Add _html_repr_ for TimeSeries

### LINKED ISSUE #2832: Provide a better __repr__ for TimeSeries/TimeSeriesMetaData
The `astropy.table.Table. __repr__` is much better than the SunPy TimeSeries one (which is not overloaded by the looks of it).

We should look at updating the one we have.

### PR DESCRIPTION
**Pull request by jiaqing23**
## PR Description

xref #2832, adding a nice `_html_repr_` for `TimeSeries`

## PR Checklist

- [ ] I have followed the guidelines in the [Contributing document](https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html)
- [ ] Changes follow the coding style of this project
- [ ] Changes have been formatted and linted
- [ ] Changes pass pytest style unit tests (and `pytest` passes).
- [ ] Changes include any required corresponding changes to the documentation
- [ ] Changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [ ] Changes have a descriptive commit message with a short title
- [ ] I have added a `Fixes #XXXX -` or `Closes #XXXX -` comment to auto-close the issue that your PR addresses

### CODE REVIEW

**Comment by nabobalis** [comment]
How does the repr look now?

**Comment by jiaqing23** [comment]
> What about the normal repr?

```
<bound method GenericTimeSeries.__repr__ of <sunpy.timeseries.sources.goes.XRSTimeSeries object at 0x000001C9594D11C0>
|-------------------------------------------------------------------------------------------------|
|TimeRange                  | Columns         | Meta                                              |
|-------------------------------------------------------------------------------------------------|
|2011-06-06T23:59:59.961999 | xrsa            | simple: True                                      |
|            to             | xrsb            | bitpix: 8                                         |
|2011-06-07T23:59:57.631999 |                 | naxis: 0                                          |
|                           |                 | extend: True                                      |
|                           |                 | date: 26/06/2012                                  |
|                           |                 | numext: 3                                         |
|                           |                 | telescop: GOES 15                                 |
|                           |                 | instrume: X-ray Detector                          |
|                           |                 | object: Sun                                       |
|                           |                 | origin: SDAC/GSFC                                 |
|                           |                 | ...                                               |
|-------------------------------------------------------------------------------------------------|

                                       xrsa          xrsb
2011-06-06 23:59:59.961999893  1.000000e-09  1.887100e-07
2011-06-07 00:00:02.008999944  1.000000e-09  1.834600e-07
2011-06-07 00:00:04.058999896  1.000000e-09  1.860900e-07
2011-06-07 00:00:06.104999900  1.000000e-09  1.808400e-07
2011-06-07 00:00:08.151999950  1.000000e-09  1.860900e-07
...                                     ...           ...
2011-06-07 23:59:49.441999912  1.000000e-09  1.624800e-07
2011-06-07 23:59:51.488999844  1.000000e-09  1.624800e-07
2011-06-07 23:59:53.538999915  1.000000e-09  1.598500e-07
2011-06-07 23:59:55.584999919  1.000000e-09  1.624800e-07
2011-06-07 23:59:57.631999850  1.000000e-09  1.598500e-07

[42177 rows x 2 columns]>
```

**Comment by nabobalis** [comment]
Can we improve the old one too?

**Comment by nabobalis** [comment]
> Do you mean the table on top?

Yeah, how does the astropy.table.Table repr look like or even the timeseries one?

**Comment by nabobalis** [comment]
I do wonder how this will work with #5834 as well.

**Comment by jiaqing23** [comment]
For `astropy.Table.table`, it looks like:
```
<bound method Table.__repr__ of <Table length=1440>
           time                 intensity      
                                  W / m2       
          object                 float64       
-------------------------- --------------------
2022-04-05 01:50:38.420147                  0.0
2022-04-05 01:51:38.420147  0.02617694830787315
2022-04-05 01:52:38.420147  0.05233595624294383
2022-04-05 01:53:38.420147  0.07845909572784494
2022-04-05 01:54:38.420147  0.10452846326765346
2022-04-05 01:55:38.420147  0.13052619222005157
2022-04-05 01:56:38.420147  0.15643446504023087
2022-04-05 01:57:38.420147  0.18223552549214744
2022-04-05 01:58:38.420147  0.20791169081775931
                       ...                  ...
2022-04-06 01:40:38.420147  -0.2588190451025236
2022-04-06 01:41:38.420147 -0.23344536385591158
2022-04-06 01:42:38.420147 -0.20791169081776192
2022-04-06 01:43:38.420147  -0.1822355254921534
2022-04-06 01:44:38.420147  -0.1564344650402332
2022-04-06 01:45:38.420147  -0.1305261922200573
2022-04-06 01:46:38.420147 -0.10452846326765551
2022-04-06 01:47:38.420147  -0.0784590957278504
2022-04-06 01:48:38.420147  -0.0523359562429456
2022-04-06 01:49:38.420147 -0.02617694830787832>
```

**Comment by jiaqing23** [comment]
Do you prefer to change the TimeSeries repr to looks similar as Table?

**Comment by nabobalis** [comment]
I'm not sure what I would like it to look like, but I don't like what it currently looks like.

**Comment by jiaqing23** [comment]
Just to confirm, you dont like both astropy.table.Table and TimeSeries?

**Comment by nabobalis** [comment]
Just the time series one. 

For now, we could leave this PR as just the nice html repr but as far as I can tell, it doesn't solve the original issue which is to fix the current normal repr.

That will need checking with https://github.com/sunpy/sunpy/pull/5834 first, since we change a lot of the underlying code.

**Comment by jiaqing23** [comment]
Sure, should I change the description etc.?

**Comment by nabobalis** [comment]
Please.

**Comment by jiaqing23** [comment]
Hi @nabobalis, I made those changes.

File: sunpy/timeseries/timeseriesbase.py [hunk 1]
```diff
@@ -123,6 +129,68 @@ def __init__(self, data, meta=None, units=None, **kwargs):
         # self._validate_meta()
         # self._validate_units()
 
+    def _text_summary(self):
+        start_time, end_time = self.time_range.start, self.time_range.end
+        center_time = self.time_range.center
+        duration = str(self.time_range.days.value) + " days"
+        columns = ", ".join(self.columns)
+
+        return textwrap.dedent("""\
```
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:138]: Can we use f strings instead of format?

File: sunpy/timeseries/timeseriesbase.py [hunk 2]
```diff
@@ -699,3 +767,10 @@ def _parse_file(cls, filepath):
             The path to the file you want to parse.
         """
         raise NoMatchError(f'Could not find any timeseries sources to parse {filepath}')
+
+
+def _figure_to_base64(fig):
```
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:772]: Is this function not used for the map html repr?

File: sunpy/timeseries/timeseriesbase.py (hunk 1 above)
Comment (jiaqing23) [reply sunpy/timeseries/timeseriesbase.py]: Fixed

File: sunpy/timeseries/timeseriesbase.py (hunk 2 above)
Comment (jiaqing23) [reply sunpy/timeseries/timeseriesbase.py]: Change it to be imported from map, maybe can be moved to "utils" but doesnt find a suitable file

Comment (nabobalis) [reply sunpy/timeseries/timeseriesbase.py]: Putting it in the base utils file seems like a decent idea?

**Comment by wafels** [comment]
See $5951

File: sunpy/timeseries/timeseriesbase.py [hunk 3]
```diff
@@ -147,7 +209,7 @@ def source(self):
 
     @property
     def observatory(self):
-        """
+        """rmat
```
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:212]:
```suggestion
        """
```

File: changelog/6032.feature.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Add `__repr_html__` for `TimeSeries`.
```
Comment (nabobalis) [inline changelog/6032.feature.rst:1]:
```suggestion
Add a basic HTML representation for `~ sunpy.timeseries.TimeSeries`.
```

File: changelog/6032.feature.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Add a basic HTML representation for `~ sunpy.timeseries.TimeSeries`.
```
Comment (nabobalis) [inline changelog/6032.feature.rst:1]:
```suggestion
Add a basic HTML representation for `~sunpy.timeseries.TimeSeries`.
```

**Comment by nabobalis** [comment]
Thanks for the PR @jiaqing23, I wanted to do some changes as I want to get this PR merged in ASAP.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**